feat: 일별/주차별/월별 수입/지출 계산 쿼리 작성

ratcomp9992 opened the pull request:
## #️⃣연관된 이슈
https://github.com/project-porko/porko-service/issues/53 기간별 수입/지출 계산 쿼리 작성

## 📝작업 내용
특정 날짜 or 특정 기간의 수입/지출 값을 QueryDSL을 사용해 계산했습니다.

[QueryDSL을 이용한 쿼리 작성](https://github.com/project-porko/porko-service/commit/c5e2eb63a7d023a9bf3c87ba19034fe1434ff9b9)

---
이후에 캘린더쪽 작업할 때 다시 수정되어야 하지만,
입출금 내역 보기 쪽에서 사용해야 하는 부분이기 때문에 우선 머지하려고 합니다.

ratcomp9992 committed:
feat(#53): 일별/주차별/월별 수입/지출 계산 쿼리 작성

- calcDailyUsedCost() : 특정 날짜 지출 금액 합
- calcDailyEarnedCost() : 특정 날짜 수입 금액 합
- calcUsedCostForPeriod() : 특정 기간 지출 금액 합
- calcEarnedCostForPeriod() : 특정 기간 수입출 금액 합

re-mora reviewed (changes requested):
History Entity가 누락된 것 같아요!
커밋에 포함시켜주세요~

ratcomp9992 committed:
feat(#53): History 도메인 추가

- (#29) 에서 몇 가지 수정
  - Type : enum 클래스 추가
  - History Entity :
       - memberId 추가
       - memo 'nullable = false' 삭제
  - SpendingCategory :
       - categoryId -> GeneratedValue로 변경
       - AllArgsConstructor 삭제
       - 생성자 추가
       - of() 에서 categoryId 삭제
